controllers/managedocs_controller.go

```diff
@@ -567,11 +568,22 @@ func (r *ManagedOCSReconciler) updateStorageClusterFromAddonParamsSecret(sc *ocs
 	addonParams := addonParamSecret.Data
 
 	sizeAsString := string(addonParams[storageClassSizeKey])
-	r.Log.Info("Requested add-on settings", storageClassSizeKey, sizeAsString)
+	enableMcgAsString := string(addonParams[enableMcgKey])
+	r.Log.Info("Requested add-on settings", storageClassSizeKey, sizeAsString, enableMcgKey, enableMcgAsString)
 	desiredDeviceSetCount, err := strconv.Atoi(sizeAsString)
 	if err != nil {
 		return fmt.Errorf("Invalid storage cluster size value: %v", sizeAsString)
 	}
+	if enableMcgAsString != "" {
+		desiredMcgEnable, err := strconv.ParseBool(enableMcgAsString)
+		if err != nil {
+			return fmt.Errorf("Invalid Enable MCG value: %v", enableMcgAsString)
+		}
+		if desiredMcgEnable {
+			r.Log.Info("Enabling MCG")
+			sc.Spec.MultiCloudGateway.ReconcileStrategy = "manage"
+		}
+	}
```

Comment:
Moving this section to end of the function causes 18 unit tests to fail since none of them are  expecting the new parameter
`enable-mcg`. I feel we can do two things
1. fix all the failing tests to include a dummy value for the new parameter 
2. Move this to a new reconcile function and make it execute just after` updateStorageClusterFromAddonParamsSecret()` is called, so the new addon won't disrupt existing tests.

Reply:
I believe it happens not because of order, but because the parameter is a non optional one (as opposed to the original code where you have introduced it as optional)

And yes, when we introduce non-optional parameters we need to make changes to both unit-tests and makefile (to make sure `make run` is working correctly) because failing on a missing parameter is the desired and expected behavior